drivers/storage/portworx/util/util.go

```diff
@@ -554,22 +556,35 @@ func GetPortworxVersion(cluster *corev1.StorageCluster) *version.Version {
 		pxVersion, err = version.NewSemver(pxVersionStr)
 		if err != nil {
 			logrus.Warnf("Invalid PX version %s extracted from image name: %v", pxVersionStr, err)
-			if pxVersionStr, exists := cluster.Annotations[AnnotationPXVersion]; exists {
-				pxVersion, err = version.NewSemver(pxVersionStr)
-				if err != nil {
-					logrus.Warnf("Invalid PX version %s extracted from annotation: %v", pxVersionStr, err)
-					pxVersionStr = getPortworxVersionFromManifestURL(manifestURL)
-					pxVersion, err = version.NewSemver(pxVersionStr)
-					if err != nil {
-						logrus.Warnf("Invalid PX version %s extracted from release manifest url: %v", pxVersionStr, err)
-					}
-				}
-			}
+			pxVersion = nil
+		} else {
+			logrus.Infof("Using version extracted from image name: %s", pxVersionStr)
```

Comment:
Let return here if we got the version, no need to check annotation or PX release manifest

Reply:
👍